### PR TITLE
[ML] adds latest record timestamp to model snapshot deprecation warning

### PR DESCRIPTION
**Pull request by benwtrent**
This adds the latest record timestamp of the deprecated model snapshot in the warning. 

This hopefully gives the user a better view into whether they would want to delete the snapshot or to upgrade it.

### CODE REVIEW

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/MlDeprecationChecker.java [hunk 1]
```diff
@@ -53,6 +54,20 @@
...
+                Locale.ROOT,
+                "model snapshot [%s] for job [%s] supports minimum version [%s] and needs to be at least [%s].",
+                modelSnapshot.getSnapshotId(),
+                modelSnapshot.getJobId(),
+                modelSnapshot.getMinVersion(),
+                Version.V_7_0_0));
+            if (modelSnapshot.getLatestRecordTimeStamp() != null) {
+                details.append(String.format(
+                    Locale.ROOT,
+                    " The model snapshot's latest record timestamp is [%s]",
```
Comment (przemekwitek) [inline x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/MlDeprecationChecker.java:67]:
```suggestion
                    ". The model snapshot's latest record timestamp is [%s]",
```

**Review by przemekwitek — APPROVED** [review]
LGTM

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/MlDeprecationChecker.java (hunk 1 above)
Comment (droberts195) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/MlDeprecationChecker.java:67]: It looks like there's already a full stop at the end of the message in line 59, so we shouldn't need another at the beginning of this extension.

I do think there's a slight formatting problem though, as the message on line 59 doesn't begin with a capital letter and this extension doesn't end with a full stop.  So we have proper sentence structure at the join but not at the very beginning or end.

We want these to look consistent with other warnings when they're displayed in the upgrade assistant in Kibana. @alisonelizabeth do you have any advice on whether these messages should have capital letters and full stops to fit in with the others that Kibana displays?

**Review by droberts195 — APPROVED** [review]
LGTM if you could just confirm that the message is in the most appropriate format to look good in the Kibana upgrade assistant.

File: x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/MlDeprecationChecker.java (hunk 1 above)
Comment (przemekwitek) [reply x-pack/plugin/deprecation/src/main/java/org/elasticsearch/xpack/deprecation/MlDeprecationChecker.java:67]: > It looks like there's already a full stop at the end of the message in line 59, so we shouldn't need another at the beginning of this extension.

Ah, right. I was looking at the wrong place.

**Comment by alisonelizabeth** [comment]
@benwtrent format looks fine with me. I also opened https://github.com/elastic/elasticsearch/issues/73089 to support a way to provide parameters to a deprecation. Feel free to add anything there that I might have missed.

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**